### PR TITLE
89 apply 95% code coverage

### LINKED ISSUE #51: Expired google oauth token pushes user to repeat tool run
If you have expired google oauth token and tryin to run C4G you will receive error on first run
And second subsequal run will be fine - you will be redirected to oauth consent screen etc...

Eliminate error and make tool run successfuly from first run in this case

### LINKED ISSUE #89: Apply 95% code coverage workflow
If code coverage is less then 95% we should fail status check and avoid merging a PR.

- [x] Add to workflow coverage checking and fail the job if coverage less then 95%
- [x] Write all tests to pass the check

Additional small tasks:
- [x] Make one workflow file `on_pull_request.yaml` to run all jobs on push commit in branch
- [x] Make one workflow file `on_push_master.yaml` to run all jobs on push commit in master

This two small tasks about simplification visibility for all workflows

### PR DESCRIPTION
**Pull request by vangogih**
## Summary

- Enforce 95% code coverage threshold in CI — added `CheckCoverageThreshold.cs` script that validates line, branch, and method coverage from Summary.json and fails the workflow if any metric drops below 95%. Integrated into `job-dotnet_test_coverage`.

- Achieve 100% standalone test coverage — added extensive unit tests for `C4GFacade`, `CodeGenerator`, `ConfigsSerializer`, `IO`, `Result`, `StackTraceUtility`, `SheetsParsingUtils`, and more. Fixed a JSON deserialization issue and an OS-specific test failure along the way.

- Restructure CI workflows into orchestrator/reusable-job pattern — replaced independent workflow files with three flow- orchestrators (`flow-on_pull_request`, `flow-on_push_master`, `flow-release`) that call job- reusable workflows. Added human-readable names to every job and step. Website build/deploy now runs conditionally only when docs change.

- Extract interfaces for testability — introduced `ICodeGenerator`, `IConfigsSerializer`, `IGoogleInteraction`, and `IIO` to enable mocking in unit tests.

- New ADRs — added code-coverage-threshold.md (enforcement rationale), ci-workflow-naming-convention.md (flow/job prefix rules), and updated ci-workflow-coordination.md to reflect the new architecture.

- Also fixed #51 with commit: 6229e481753bad4a15bde0aad896dd93bd9bd79d

### CODE REVIEW

**Comment by vangogih** [comment]
@cursoragent  review

File: .github/workflows/scripts/CheckCoverageThreshold.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env dotnet run
+
```
[the remark this answers is not in the document]
Comment (vangogih) [reply .github/workflows/scripts/CheckCoverageThreshold.cs:2]: Not an issue, we just follow official doc

File: C4G.Unity/Assets/C4G/Core/Scripts/SheetsParsing/SheetsParsingUtils.cs [hunk 2]
```diff
@@ -205,27 +205,26 @@ internal static Result<List<ConfigFrame>, string> ParseConfigFrames(string sheet
...
                             return Result<List<ConfigFrame>, string>.FromError($"{errorPrefix}Cell [{rowIndex + 1}][{columnIndex + 1}] '{cellText}' has invalid name");
 
-                        int matches = 0;
+                        bool found = false;
                         for (int configFrameIndex = configFrameStarts.Count - 1; configFrameIndex >= 0; --configFrameIndex)
                         {
                             ConfigFrame configFrame = configFrameStarts[configFrameIndex];
-                            if (rowIndex >= configFrame.StartRowIndex && columnIndex >= configFrame.StartColumnIndex)
+                            if (rowIndex >= configFrame.StartRowIndex)
+                            if (columnIndex >= configFrame.StartColumnIndex)
```
Comment (alexboost222) [inline C4G.Unity/Assets/C4G/Core/Scripts/SheetsParsing/SheetsParsingUtils.cs]: Any reasons why we should change this behaviour?

File: C4G.Unity/Assets/C4G/Tests/Editor/Unity/ConfigsSerialization/ConfigsSerializerTests.EdgeCaseTests.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+			public void SerializeParsedConfigs_DuplicateSheetName_ReturnsError()
+			{
+				var config1 = new ParsedConfig("Dup", new List<ParsedPropertyInfo>(), new List<List<string>>());
+				var config2 = new ParsedConfig("Dup", new List<ParsedPropertyInfo>(), new List<List<string>>());
+
+				Result<string, string> result = _configsSerializer.SerializeParsedConfigsAsJsonObject(
+					new List<ParsedConfig> { config1, config2 }, _parsersByName);
+
+				Assert.IsFalse(result.IsOk);
+				Assert.That(result.Error, Does.Contain("Duplicate"));
```
Comment (alexboost222) [inline C4G.Unity/Assets/C4G/Tests/Editor/Unity/ConfigsSerialization/ConfigsSerializerTests.EdgeCaseTests.cs:32]: I guess we shoulnt check error texts because it is still too volatile and such tests would often break + we will switch to custom exceptions soon

File: C4G.Unity/Assets/C4G/Tests/Editor/Unity/SheetsParsing/SheetsParsingUtilsDirectTests.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
+using System.Collections.Generic;
+using C4G.Core.SheetsParsing;
+using C4G.Core.Utils;
+using NUnit.Framework;
+
+namespace C4G.Tests.Editor.Unity.SheetsParsing
+{
+	[TestFixture]
+	public class SheetsParsingUtilsDirectTests
```
Comment (alexboost222) [inline C4G.Unity/Assets/C4G/Tests/Editor/Unity/SheetsParsing/SheetsParsingUtilsDirectTests.cs:9]: I guess it would be better (and more obvious too) to test SheetsParsingUtils by directly calling desired methods

**Review by alexboost222 — CHANGES_REQUESTED** [review]
Very good job! There are several comments to fix and then it would be ready for merge!

File: C4G.Unity/Assets/C4G/Tests/Editor/Unity/SheetsParsing/SheetsParsingUtilsDirectTests.cs (hunk 4 above)
Comment (vangogih) [reply C4G.Unity/Assets/C4G/Tests/Editor/Unity/SheetsParsing/SheetsParsingUtilsDirectTests.cs:9]: Sorry didn't get the idea. I think I do so here.

Can you write a small example?

File: C4G.Unity/Assets/C4G/Core/Scripts/SheetsParsing/SheetsParsingUtils.cs (hunk 2 above)
Comment (vangogih) [reply C4G.Unity/Assets/C4G/Core/Scripts/SheetsParsing/SheetsParsingUtils.cs]: Reverted the changes 👍🏻

Comment (vangogih) [reply C4G.Unity/Assets/C4G/Core/Scripts/SheetsParsing/SheetsParsingUtils.cs]: Removed dead code in ParseConfigFrames that was impossible to cover:

`rowIndex >= configFrame.StartRowIndex` — always true because the outer loop scans rows top-to-bottom, so all accumulated starts have `StartRowIndex <= rowIndex` by the time an end cell is reached.

`if (++matches > 1)` — unreachable because the first geometric match either breaks (name matched) or returns an error (name mismatched). The loop never continues to a second match.

Replaced int matches counter with bool found — same edge cases, zero dead code, 100% line + branch coverage.

File: C4G.Unity/Assets/C4G/Tests/Editor/Unity/ConfigsSerialization/ConfigsSerializerTests.EdgeCaseTests.cs (hunk 3 above)
Comment (vangogih) [reply C4G.Unity/Assets/C4G/Tests/Editor/Unity/ConfigsSerialization/ConfigsSerializerTests.EdgeCaseTests.cs:32]: I do that in tests because we have to be sure what kind of error we call to be sure that we covered correct execution branch. Easy to fix because mismatch will be shown on logs.

**Comment by alexboost222** [comment]
I got reasonable answers to the comments, approved

**Review by alexboost222 — APPROVED** [review]
Legit!
